content/how-to/bpmn/trigger-workflow.md

```diff
@@ -0,0 +1,145 @@
+---
+title: 'Trigger workflows'
+date: '2024-04-24T19:35:09+03:00'
+categories: ["how-to"]
+description: How to Trigger Workflow
+weight: 150
+menu:
+  main:
+    parent: howto-bpmn
+    identifier:
+---
+
+
+You also have multiple ways to start, or _trigger_, a BPMN workflow.
+The best choice of trigger depends on the context of the event and the system that initiates the workflow.
+
+{{% reusable/bpmn-triggers %}}
+
+## Start a workflow from an API
+
+When a workflow has the default start event, trigger it through an API call.
+For example, an API trigger may originate from a custom frontend application or a developer's local machine.
+
+### Synchronous and asynchronous API triggers
+
+To start BPMN workflows, Rhize has two API operations:
+- `createAndRunBPMNSync` starts a workflow and waits for the process to complete or abort (synchronous).
+- `createAndRunBpmn` starts a workflow and does not wait for the response (asynchronous).
+
+The synchronous call does not execute further operations until it receives a response from the Rhize server.
+On the other hand, the asynchronous operation frees up the call system to do more work, no matter whether the workflow runs correctly.
+
+The call syntax for the each operation is almost identical.
+For example, compare the syntax for these calls:
+
+{{% tabs %}}
+{{% tab "Synchronous" %}}
+```gql
+mutation SynchronousCall($createAndRunBpmnSyncId: String!, $version: String ) {
+  createAndRunBpmnSync(id: $createAndRunBpmnSyncId, version: $version) {
+    id
+    jobState
+    customResponse
+  }
+}
+```
+{{% /tab %}}
+
+{{% tab "Async" %}}
+```gql
+mutation AsyncCall{
+  createAndRunBPMNSync{
+    id
+    jobState
+    customResponse
+  }
+}
+```
+{{% /tab %}}
+{{% /tabs %}}
+
+The responses for these calls have two differences:
+- For synchronous calls, the returned `JobState` should be a finished value (such as `COMPLETED` or `ABORTED`). Asynchronous calls likely return an in-progress status, such as `RUNNING`.
+- Only the synchronous call receives data in the `customResponse`. For details, refer to the next section.
+
+### `CustomResponse`
+
+The `customResponse` is a special variable to return data in the response to clients that run `createAndRunBPMNSync` operations.
+
+You can set the `customResponse` in any [element]({{< relref "/how-to/bpmn/bpmn-elements" >}}) that has an `Output` or `Input response` parameter.
+It can use any data from the {{< abbr "process variable context" >}}, including variables added on the fly.
+Its format can use JSON or JSONata.
```

Comment:
The format will always be JSON. Some part of the process must create a variable in the context called `customResponse` and that variable is returned as a JSON string.